src/tainting/Dataflow_tainting.ml

```diff
@@ -902,8 +949,10 @@ let check_function_signature env fun_exp args_taints =
                  Some
                    (Taints.singleton
                       { orig = Src { src with call_trace }; tokens = [] })
-             | T.ArgToReturn (argpos, tokens, _return_tok) ->
-                 let* arg_taints = taints_of_arg argpos in
+             | T.ArgToReturn (arg, tokens, _return_tok) ->
+                 let* arg_taints =
+                   taints_of_sig_arg env fparams args args_taints arg
```

Comment:
I'm surprised we didn't have to do this before. Does this also improve our field sensitivity in other ways than just the getX case?

Reply:
Note that this is an incomplete solution, it doesn't handle all cases, but it's mean to handle the most basic ones. It does work for more than just getters, I have tried it with Python, and I also managed to get something working for JS/TS. But I feel it needs some testing and for now I feel more comfortable restricting it to Java and `getX`.